backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -0,0 +1,211 @@
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Trait of parameters passed to a StandardCacheHitCopyingActor.
+  */
+trait StandardCacheHitCopyingActorParams {
+  def jobDescriptor: BackendJobDescriptor
+
+  def backendInitializationDataOption: Option[BackendInitializationData]
+
+  def serviceRegistryActor: ActorRef
+  
+  def ioActor: ActorRef
+
+  def configurationDescriptor: BackendConfigurationDescriptor
+}
+
+/** A default implementation of the cache hit copying params. */
+case class DefaultStandardCacheHitCopyingActorParams
+(
+  override val jobDescriptor: BackendJobDescriptor,
+  override val backendInitializationDataOption: Option[BackendInitializationData],
+  override val serviceRegistryActor: ActorRef,
+  override val ioActor: ActorRef,
+  override val configurationDescriptor: BackendConfigurationDescriptor
+) extends StandardCacheHitCopyingActorParams
+
+object StandardCacheHitCopyingActor {
+  type DetritusMap = Map[String, Path]
+  type PathPair = (Path, Path)
+  
+  sealed trait StandardCacheHitCopyingActorState
+  case object Idle extends StandardCacheHitCopyingActorState
+  case object WaitingForCopyResponses extends StandardCacheHitCopyingActorState
+
+  case class StandardCacheHitCopyingActorData(copyCommandsToWaitFor: Set[IoCopyCommand],
+                                                   copiedJobOutputs: CallOutputs,
+                                                   copiedDetritus: DetritusMap,
+                                                   returnCode: Option[Int]
+                                                  ) {
+    def remove(copyCommand: IoCopyCommand) = copy(copyCommandsToWaitFor = copyCommandsToWaitFor filterNot { _ == copyCommand })
+  }
+}
+
+class DefaultStandardCacheHitCopyingActor(standardParams: StandardCacheHitCopyingActorParams) extends StandardCacheHitCopyingActor(standardParams) with DefaultIoCommandBuilder
+
+/**
+  * Standard implementation of a BackendCacheHitCopyingActor.
+  */
+abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHitCopyingActorParams)
+  extends FSM[StandardCacheHitCopyingActorState, Option[StandardCacheHitCopyingActorData]] with JobLogging with StandardCachingActorHelper with IoClientHelper { this: IoCommandBuilder =>
+
+  override lazy val jobDescriptor: BackendJobDescriptor = standardParams.jobDescriptor
+  override lazy val backendInitializationDataOption: Option[BackendInitializationData] = standardParams.backendInitializationDataOption
+  override lazy val serviceRegistryActor: ActorRef = standardParams.serviceRegistryActor
+  override lazy val configurationDescriptor: BackendConfigurationDescriptor = standardParams.configurationDescriptor
+  
+  lazy val destinationCallRootPath: Path = jobPaths.callRoot
+  lazy val destinationJobDetritusPaths: Map[String, Path] = jobPaths.detritusPaths
+  lazy val ioActor = standardParams.ioActor
+  
+  startWith(Idle, None)
+  
+  context.become(ioReceive orElse receive)
```

Comment:
I wasn't suggesting we not do it; it enables bringing in those other behaviors as something like cross-cutting concerns.  But it did surprise me.  😄